### DISCUSSION TITLE
AIS Catcher TCP feed

### DISCUSSION THREAD

**Discussion by lucasbeumer** [jvde-github/AIS-catcher#322] [Ideas]
Hello !
I am a retired Dutch telecom engineer and fascinated about AIS applications
I tested AIS Catcher and it is an impressive application.
I want to integrate the data from one AIS Catcher feed station (for instance station 1621) into my ECDIS application. My ECDIS only accepts TCP connection.
Would it be possible that AIS catcher has an TCP server to make the data available as TCP streams?
For instance, AIS catcher transmits data from station 1621 on port 1621, transmits data from station 1622 on port 1622. ?? 
In this way I could select data from one or more feed stations into my ECDIS.
THANKS
Lucas

**Comment by Phil353556** [comment]
Hi,
I'm just a user, but yes TCP stream is possible via ais-catcher
see : https://github.com/jvde-github/AIS-catcher
_"forwarded via UDP/HTTP/TCP."_ 
and 
_"-P xxx.xx.xx.xx yyy - TCP destination address and port (default: off)]"_
and paragraph:
_"Forwarding AIS messages over UDP and TCP"_

just need to add a line in the config.cmd or json file
-P  xxx.xxx.xxx.xxx  yyy 
   xxx.xxx.xxx.xxx  the IP of the system that must receive the TCP stream
   yyy   the TCP port number to connect to 
   
   hope this helps
   Phil

**Comment by jvde-github** [comment]
Hello Lucas!

Thanks for reaching out. At the moment it is not possible to read back data
from ais.

It is possible though with your local AIS-catcher receiver. if you run
with, say, -S 5012, the program starts a TCP server at port 5012. But I
guess your idea is to receive the AIS data from other feeders to the site,
right? I have not thought about that to be honest.

Thanks,
Jasper

Op wo 30 okt 2024 om 11:19 schreef Lucas Beumer PA0NFN <
***@***.***>:

**Comment by lucasbeumer** [comment]
Hello Jasper

 

Thank you for coming back , just a few hours after posting my request !!   Are you  Dutch ?? , looking at ur name? 

 

Yes  I tested with my  local AIS-catcher -S 5012, and works fine.



Yes correct,, it is the  idea  to receive the AIS data received from any selected feeder (via AIS-catcher)  back to me as incoming  TCP stream.

In this way I will forward the received nmea strings (raw data) to my Electronic chart application where I would see a vessel sailing in a channel or along the coast.

My ECDIS allows to set alarm when a vessel enters a zone or exceed a set speed. With the AIS-catcher data I could use this as a test source.

 

Thank you so much !

73

 

Lucas

**Comment by jvde-github** [comment]
All clear, I will keep it in mind when expanding the system. 

Cheers, Jasper

ps from NL indeed

**Reply by lucasbeumer** [reply]
Hello Jasper,   Thank you for all the hard work .  Good success when expanding the system, Cheers Lucas

**Comment by LucasGuyancourt** [comment]
Hello Jasper

I am coming back to my old request to receive AIS data from other feeders by TCP connection
In this way I am able to use AIS catcher raw AIS data with my ECDIS application
Any progress on this topic?

Best regards
Lucas
73
